src/data.jl

```diff
@@ -847,17 +847,7 @@ h5type(f::JLDFile, ::Type{T}, x) where {T<:DataType} =
     h5fieldtype(f, DataType, typeof(x), Val{true})
 odr(::Type{T}) where {T<:DataType} = DataTypeODR()
 
-function typename(T::DataType)
-    m = T.name.module
-    tn = Symbol[nameof(m)]
-    while m != parentmodule(m)
-        push!(tn, nameof(m))
-        m = parentmodule(m)
-    end
-    reverse!(tn)
-    push!(tn, T.name.name)
-    join(tn, ".")
-end
+typename(T::DataType) = join([fullname(T.name.module)..., T.name.name], ".")
```

Comment:
```suggestion
typename(T::DataType) = join((fullname(T.name.module)..., T.name.name), '.')
```

The timings (Julia 1.2.0) should be the same (I cannot see the difference with `@time`), but the tuple allocates less:
```
julia> @time join([fullname(MCLResult.name.module)..., MCLResult.name.name], ".")
  0.000055 seconds (13 allocations: 544 bytes)
"Clustering.MCLResult"

julia> @time join((fullname(MCLResult.name.module)..., MCLResult.name.name), '.')
  0.000058 seconds (13 allocations: 480 bytes)
"Clustering.MCLResult"
```

Or even
```julia
function typename(T::DataType)
    s = IOBuffer()
    join(s, fullname(T.name.module), '.')
    print(s, '.')
    print(s, T.name.name)
    return String(resize!(s.data, s.size))
end
```

This one allocates less, and (while the timings fluctuate a lot) is definitely faster:
```
julia> @time typename(MCLResult)
  0.000016 seconds (8 allocations: 368 bytes)
"Clustering.MCLResult"
```

Reply:
Done. FYI I prefer to use `BenchmarkTools.@btime` instead of `@time`.